expr/func_constantline_test.go

```diff
@@ -0,0 +1,130 @@
+package expr
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/test"
+)
+
+func TestConstantLineSimple(t *testing.T) {
+	testConstantLineWrapper("simple", 1, 0, 100, t)
+}
+
+func testConstantLineWrapper(name string, value float64, from uint32, to uint32, t *testing.T) {
+	testConstantLine(name, value, from, to, makeConstantLineSeries(value, from, to), t)
+}
+
+func makeConstantLineSeries(value float64, from uint32, to uint32) ([]models.Series) {
+	series := []models.Series{
+		{
+			Target:		 fmt.Sprintf("%g", value),
+			QueryPatt:   fmt.Sprintf("%g", value),
+			Datapoints: []schema.Point{
+				{Val: value, Ts: from},
+				{Val: value, Ts: from + uint32((to-from)/2.0)},
+				{Val: value, Ts: to},
+			},
+		},
+	}
+
+	return series
+}
+
+func testConstantLine(name string, value float64, from uint32, to uint32, out []models.Series, t *testing.T) {
+	f := NewConstantLine()
+	f.(*FuncConstantLine).value  = value
+	f.(*FuncConstantLine).from = from
+	f.(*FuncConstantLine).to = to
+	gots, err := f.Exec(make(map[Req][]models.Series))
+
+	if err != nil {
+		t.Fatalf("case %q: err should be nil. got %q", name, err)
+	}
+	if len(gots) != len(out) {
+		t.Fatalf("case %q: isNonNull len output expected %d, got %d", name, len(out), len(gots))
+	}
+
+	for i, g := range gots {
```

Comment:
Can you use [equalOutput](https://github.com/grafana/metrictank/blob/ed6257fb34ac4ef7d09bae0c534c694fb64ada96/expr/test.go#L12) like [round](https://github.com/grafana/metrictank/blob/master/expr/func_round_test.go#L281) does?